rosbridge_library/src/rosbridge_library/internal/subscribers.py

```diff
@@ -120,7 +120,8 @@ def __init__(self, topic, client_id, callback, node_handle, msg_type=None, raw=F
         )
 
         infos = node_handle.get_publishers_info_by_topic(topic)
-        if any(pub.qos_profile.durability == DurabilityPolicy.TRANSIENT_LOCAL for pub in infos):
+
+        if all(pub.qos_profile.durability == DurabilityPolicy.TRANSIENT_LOCAL for pub in infos):
```

Comment:
I forgot to take into account a situation where there are no publishers in which case this statement will be true. This should fix it:
```suggestion
        if len(infos) > 0 and all(
            pub.qos_profile.durability == DurabilityPolicy.TRANSIENT_LOCAL for pub in infos
        ):
```